### PR TITLE
migrate from Travis CI to GitHub Actions

### PR DESCRIPTION
**Pull request by AurelienGasser**
It's all in the title!

Builds should be triggered for

- pushes to master
- pushes to PRs (regardless of target branch)

Build times are comparable as with Travis (~1 min)

image

Once we move to a more recent version of Go, the "Install Go" step should be faster (version 1.12 is not cached)

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+name: Build and test Go
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: "*"
```
Comment (AlexandrePicosson) [inline .github/workflows/go.yml]: Here you can target only master, this has the benefit of running the CI only once on the code on a scenario like this:
```
master ---------------
                 /
pr1 ------------/
         /
pr2 ----/
```

With your solution you test on `pr2 -> pr1` and `pr1 -> master` and changing `branches:` to `master` you would build only on `pr1 -> master`.

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        with:
+          path: ~/go/pkg/mod
+          key: build-${{ hashFiles('**/go.sum') }}
+      - name: Build
+        run: go build -v
+        working-directory: ./chaincode
+      - name: Test
+        run: go test -v ./...
+        working-directory: ./chaincode
+      - name: Lint
```
Comment (AlexandrePicosson) [inline .github/workflows/go.yml]: You can run the lint first in this testing pipeline, that way we fail early without running the potentially costly operation of building the go package (build takes `13s` vs `2s` for the lint) if the code does not comply with the lint rules.

File: .github/workflows/go.yml (hunk 1 above)
Comment (AurelienGasser) [reply .github/workflows/go.yml]: This actually intentional. Unit testing `pr2 -> pr1` is also useful, to catch errors early, no?

Comment (AlexandrePicosson) [reply .github/workflows/go.yml]: This was not done previously with travis but if this is a feature you would like to have then we can revert the change and put it back like it was before.

**Review by AlexandrePicosson — APPROVED** [review]

**Review by mblottiere — APPROVED** [review]

File: .github/workflows/go.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  build:
+    name: Build and test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out source code
+        uses: actions/checkout@v2
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.12.x
```
Comment (inelgnu) [inline .github/workflows/go.yml:22]: Maybe not related to this PR, but I think we should consider using a more recent version of Go.

**Review by inelgnu — COMMENTED** [review]
LGTM, minor comment on version that could be treated seperately

**Review by inelgnu — APPROVED** [review]

File: .github/workflows/go.yml (hunk 3 above)
Comment (AurelienGasser) [reply .github/workflows/go.yml:22]: Agreed. I used 1.12 as this is the documented/tested go version for the chaincode (in the README and go.mod), but we should move to 1.15.

**Pull request merged**